vars/deploy.groovy

```diff
@@ -0,0 +1,149 @@
+    /** The stage name. If the stage name is not provided, it will be taken from the environment variable 'STAGE_NAME'.*/
+    'stage',
+    /** Defines the deployment type */
+    'enableZeroDowntimeDeployment',
+    /** The source file to deploy to neo*/
+    'source'
+])
+
+@GenerateDocumentation
+void call(parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        def stageName = parameters.stage ?: env.STAGE_NAME
+        def enableZeroDowntimeDeployment = parameters.enableZeroDowntimeDeployment ?: false
+
+        def script = checkScript(this, parameters) ?: this
+        def utils = parameters.utils ?: new Utils()
+        def jenkinsUtils = parameters.jenkinsUtils ?: new JenkinsUtils()
+
+        Map config = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        utils.pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'stage',
+            stepParam1: stageName,
+            stepParamKey2: 'enableZeroDowntimeDeployment',
+            stepParam2: enableZeroDowntimeDeployment
+        ], config)
+
+        def index = 1
+        def deployments = [:]
+        def deploymentType
+        def deployTool
+
+        if (config.cfTargets) {
+
+            deploymentType = DeploymentType.selectFor(CloudPlatform.CLOUD_FOUNDRY, enableZeroDowntimeDeployment).toString()
+            deployTool = script.commonPipelineEnvironment.configuration.isMta ? 'mtaDeployPlugin' : 'cf_native'
+
+            for (int i = 0; i < config.cfTargets.size(); i++) {
+
+                def target = config.cfTargets[i]
+
+                Closure deployment = {
+
+                    utils.unstash(stageName)
+
+                    cloudFoundryDeploy(
+                        script: script,
+                        juStabUtils: utils,
+                        jenkinsUtilsStub: jenkinsUtils,
+                        deployType: deploymentType,
+                        cloudFoundry: target,
+                        mtaPath: script.commonPipelineEnvironment.mtarFilePath,
+                        deployTool: deployTool
+                    )
+
+                    utils.stash(stageName)
+                }
+                runDeployment(deployments, deployment, index, script, stageName)
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+
+        if (config.neoTargets) {
+
+            deploymentType = DeploymentType.selectFor(CloudPlatform.NEO, enableZeroDowntimeDeployment)
+
+            def source
+            if (config.source) {
+                source = config.source
+            } else {
+                def pom = readMavenPom file: 'application/pom.xml'
+                source = "application/target/${pom.getArtifactId()}.${pom.getPackaging()}"
```

Comment:
I think that is something specific to our pipeline and can be removed.

Reply:
Ok. If you don't need it I will remove it.